cmd/mavenExecute.go

```diff
@@ -4,29 +4,56 @@ import (
 	"github.com/SAP/jenkins-library/pkg/command"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/maven"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"io/ioutil"
+	"strings"
 
 	"github.com/SAP/jenkins-library/pkg/telemetry"
 )
 
-func mavenExecute(config mavenExecuteOptions, telemetryData *telemetry.CustomData) string {
-	c := command.Command{}
+func mavenExecute(config mavenExecuteOptions, _ *telemetry.CustomData) {
+	runner := command.Command{}
+	err := runMavenExecute(config, &runner)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
 
+func runMavenExecute(config mavenExecuteOptions, runner execRunner) error {
 	options := maven.ExecuteOptions{
 		PomPath:                     config.PomPath,
 		ProjectSettingsFile:         config.ProjectSettingsFile,
 		GlobalSettingsFile:          config.GlobalSettingsFile,
 		M2Path:                      config.M2Path,
-		Goals:                       config.Goals,
-		Defines:                     config.Defines,
-		Flags:                       config.Flags,
+		Goals:                       splitAndTrimParams(config.Goals),
+		Defines:                     splitAndTrimParams(config.Defines),
+		Flags:                       splitAndTrimParams(config.Flags),
 		LogSuccessfulMavenTransfers: config.LogSuccessfulMavenTransfers,
 		ReturnStdout:                config.ReturnStdout,
 	}
 
-	output, err := maven.Execute(&options, &c)
-	if err != nil {
-		log.Entry().WithError(err).Fatal("step execution failed")
+	output, err := maven.Execute(&options, runner)
+	if err == nil && output != "" {
+		err = ioutil.WriteFile(".pipeline/maven_output.txt", []byte(output), 0644)
 	}
+	return err
```

Comment:
if  `err == nil`&& output =="" no output file is created but also no error is returned and the execution is successful. I would always expect a file is written if I specify ReturnStdout and the execution was successful, even if it empty.

Reply:
Good catch.